### PR TITLE
Mf 463 patient search 

### PR DESCRIPTION
**Pull request by donaldkibet**
### Description

1. Setup repo structure
2. Implemented search functionality 

### Screenshoot

mf-463

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,76 @@
+{
+  "name": "@openmrs/esm-patient-management",
+  "version": "1.0.0",
```
Comment (FlorianRappl) [inline package.json:3]: Should be dropped - only mention version in lerna.json

File: packages/esm-patient-search-app/src/patient-search/patient-search.component.tsx [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    'patientIdentifier:(uuid,identifier),' +
+    'person:(gender,age,birthdate,birthdateEstimated,personName,display),' +
+    'attributes:(value,attributeType:(name)))';
+  const searchTimeout = 300;
+  const resultsPerPage = 5;
+
+  const [searchTerm, setSearchTerm] = React.useState('');
+  const [emptyResult, setEmptyResult] = React.useState(false);
+  const [searchResults, setSearchResults] = React.useState<Array<SearchedPatient>>([]);
+  const [pagedResults, setPagedResults] = React.useState<Array<SearchedPatient>>([]);
```
Comment (FlorianRappl) [inline packages/esm-patient-search-app/src/patient-search/patient-search.component.tsx:31]: For the pagination we have a hook that we used in patient-chart.

Let's see if the hook can be (or is already) moved into esm-core (into esm-react-utils to be more specific).

**Review by FlorianRappl — COMMENTED** [review]
LGTM! Great work!

File: packages/esm-patient-search-app/src/patient-search/patient-search.component.tsx (hunk 2 above)
Comment (donaldkibet) [reply packages/esm-patient-search-app/src/patient-search/patient-search.component.tsx:31]: Moved this hook `usePagination` from [@openmrs/esm-common-lib](https://github.com/openmrs/openmrs-esm-patient-chart/tree/master/packages/esm-patient-common-lib) to `esm-react-utils`

Comment (FlorianRappl) [reply packages/esm-patient-search-app/src/patient-search/patient-search.component.tsx:31]: Awesome - thanks!

**Review by FlorianRappl — APPROVED** [review]
LGTM!

**Pull request merged**